pkg/apiserver/apiserver.go

```diff
@@ -0,0 +1,150 @@
+)
+
+const (
+	Name = "theia-manager-api"
+	// authenticationTimeout specifies a time limit for requests made by the authorization webhook client
+	// The default value (10 seconds) is not long enough as defined in
+	// https://pkg.go.dev/k8s.io/apiserver@v0.21.0/pkg/server/options#NewDelegatingAuthenticationOptions
+	// A value of zero means no timeout.
+	authenticationTimeout = 0
+)
+
+var (
+	// Scheme defines methods for serializing and deserializing API objects.
+	scheme = runtime.NewScheme()
+	// Codecs provides methods for retrieving codecs and serializers for specific
+	// versions and content types.
+	codecs = serializer.NewCodecFactory(scheme)
+	// ParameterCodec defines methods for serializing and deserializing url values
+	// to versioned API objects and back.
+	parameterCodec = runtime.NewParameterCodec(scheme)
+	// #nosec G101: false positive triggered by variable name which includes "token"
+	TokenPath = "/var/run/antrea/apiserver/loopback-client-token"
+)
+
+func init() {
+	intelligenceinstall.Install(scheme)
+	metav1.AddToGroupVersion(scheme, schema.GroupVersion{Version: "v1"})
+}
+
+type theiaManagerAPIServer struct {
+	GenericAPIServer *genericapiserver.GenericAPIServer
+}
+
+func (s *theiaManagerAPIServer) Run(stopCh <-chan struct{}) error {
+	if s == nil {
+		return fmt.Errorf("nil s")
+	}
+	p := s.GenericAPIServer
+	if p == nil {
+		return fmt.Errorf("nil")
+	}
+	return p.PrepareRun().Run(stopCh)
+}
+
+func newConfig(bindPort int) (*genericapiserver.CompletedConfig, error) {
+	secureServing := genericoptions.NewSecureServingOptions().WithLoopback()
+	authentication := genericoptions.NewDelegatingAuthenticationOptions()
+
+	// Set the PairName but leave certificate directory blank to generate in-memory by default.
+	secureServing.ServerCert.CertDirectory = ""
+	secureServing.ServerCert.PairName = Name
+	secureServing.BindAddress = net.IPv4zero
+	secureServing.BindPort = bindPort
+
+	authentication.WithRequestTimeout(authenticationTimeout)
+
+	if err := secureServing.MaybeDefaultWithSelfSignedCerts("localhost", nil, []net.IP{net.ParseIP("127.0.0.1"), net.IPv6loopback}); err != nil {
+		return nil, fmt.Errorf("error creating self-signed certificates: %v", err)
+	}
+	serverConfig := genericapiserver.NewConfig(codecs)
+	if err := secureServing.ApplyTo(&serverConfig.SecureServing, &serverConfig.LoopbackClientConfig); err != nil {
+		return nil, err
+	}
+	if err := authentication.ApplyTo(&serverConfig.Authentication, serverConfig.SecureServing, nil); err != nil {
+		return nil, err
+	}
+
+	// TODO (wshaoquan): use native k8s role-based auth
+	serverConfig.Authorization = genericapiserver.AuthorizationInfo{
+		Authorizer: authorizerfactory.NewAlwaysAllowAuthorizer()}
+	if err := os.MkdirAll(path.Dir(TokenPath), os.ModeDir); err != nil {
+		return nil, fmt.Errorf("error when creating dirs of token file: %v", err)
+	}
+	if err := ioutil.WriteFile(TokenPath, []byte(serverConfig.LoopbackClientConfig.BearerToken), 0600); err != nil {
```

Comment:
Do we need to remove ioutil in this PR as well?

Reply:
done